pkg/daemon/daemon.go

```diff
@@ -496,16 +496,21 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 			return err
 		}
 	}
+
+	if dn.nodeHasAnnotation(annoKey, annoDrainRequired) {
+		glog.Info("nodeStateSyncHandler(): waiting for drain")
+		return nil
+	}
+
 	if reqDrain {
 		if !dn.isNodeDraining() {
 			if !dn.disableDrain {
-				ctx, cancel := context.WithCancel(context.TODO())
-				defer cancel()
+				glog.Infof("nodeStateSyncHandler(): apply 'Drain_Required' label for node")
 
-				glog.Infof("nodeStateSyncHandler(): get drain lock for sriov daemon")
-				done := make(chan bool)
-				go dn.getDrainLock(ctx, done)
-				<-done
+				if err := dn.applyDrainRequired(); err != nil {
+					return err
+				}
+				return nil
```

Comment:
Here we return and not continue to the OpenShift part. Seems that it can change the logic, no?